layersvt/screenshot.cpp

```diff
@@ -364,18 +366,126 @@ static void writePPM(const char *filename, VkImage image1) {
     // Gather incoming image info and check image format for compatibility with
     // the target format.
     // This function supports both 24-bit and 32-bit swapchain images.
-    VkFormat const target32bitFormat = VK_FORMAT_R8G8B8A8_UNORM;
-    VkFormat const target24bitFormat = VK_FORMAT_R8G8B8_UNORM;
     uint32_t const width = imageMap[image1]->imageExtent.width;
     uint32_t const height = imageMap[image1]->imageExtent.height;
     VkFormat const format = imageMap[image1]->format;
     uint32_t const numChannels = vk_format_get_channel_count(format);
-    if ((vk_format_get_compatibility_class(target24bitFormat) != vk_format_get_compatibility_class(format)) &&
-        (vk_format_get_compatibility_class(target32bitFormat) != vk_format_get_compatibility_class(format))) {
+
+    if ((3 != numChannels) && (4 != numChannels)) {
         assert(0);
         return;
     }
-    if ((3 != numChannels) && (4 != numChannels)) {
+
+    // Initial dest format is undefined as we will look for one
+    VkFormat destformat = VK_FORMAT_UNDEFINED;
+
+    //Get users request is specific color space format required
+    const char *vk_screenshot_format = local_getenv(env_var_format);
```

Comment:
Parse the env var once, sometime during initiliazation, rather than each time a screenshot is generated.